task-sdk/src/airflow/sdk/execution_time/context.py

```diff
@@ -397,7 +399,29 @@ def add(self, asset: Asset | AssetRef, extra: dict[str, Any] | None = None) -> N
         self.asset_alias_events.append(event)
 
 
-class OutletEventAccessors(_AssetRefResolutionMixin, Mapping[Union[Asset, AssetAlias], OutletEventAccessor]):
+class _AssetEventAccessorsMixin(Generic[T]):
+    def for_asset(self, *, name: str | None = None, uri: str | None = None) -> T:
+        if name and uri:
+            return self[Asset(name=name, uri=uri)]
+        elif name:
+            return self[Asset.ref(name=name)]
+        elif uri:
+            return self[Asset.ref(uri=uri)]
+
+        raise ValueError("name and uri cannot both be None")
+
+    def for_asset_alias(self, *, name: str) -> T:
+        return self[AssetAlias(name=name)]
+
+    def __getitem__(self, key: Asset | AssetAlias | AssetRef) -> T:
+        raise NotImplementedError
```

Comment:
Not sure if matters, but should this include a message to say `use for_asset/for_asset_alias`?

Reply:
Oh, this is just for typing essentially isn't it?

Reply:
yep, it's for typing. Without it, mypy raises lots of warnings 😱